### ISSUE TITLE
`Master.wait()` is broken when `uid`s are specified

### ISSUE THREAD

**Issue by andre-merzky** [radical-cybertools/radical.pilot#2766]
```
                n = stats['ACTIVE'] + stats['DONE'] + stats['FAILED']
```
should be 
```
                n = len(stats['ACTIVE']) + len(stats['DONE']) + len(stats['FAILED'])
```
Also, keys need to be checked also (or use `defaultdict(list)`)

**Comment by eirrgang** [comment]
Duplicate of #2652 ?

**Comment by andre-merzky** [comment]
An RP ticket is now open for this on radical-cybertools/radical.pilot/issues/2776

**Comment by andre-merzky** [comment]
Fixed by PR #2780
